File: testing/unit-tests/pkmntest/csharp/Gen1PokemonTests.cs
/*
 * Copyright (c) 2017-2018 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

using System;
using NUnit.Framework;

namespace PKMNTest
{

public class Gen1PokemonTests
{
    public static void PokemonTest(
        PKMN.Game game,
        PKMN.Species species
    )
    {
        PKMN.Pokemon pokemon = new PKMN.Pokemon(species, game, "", 30);

        PokemonTestParams testParams = new PokemonTestParams(
                                               PKMN.Ball.GREAT_BALL,
                                               new PKMN.Ball[]{PKMN.Ball.GREAT_BALL},
                                               PKMN.Item.POTION,
                                               new PKMN.Item[]{PKMN.Item.POTION},
                                               "Special",
                                               new string[]{"Route 1"},
                                               new string[]{"Route 1"},
                                               new string[]{"Slash", "Flamethrower", "Tail Whip", "Fire Blast"},
                                               new string[]{"Return"},
                                               new PKMN.Game[]{PKMN.Game.RED},
                                               new PKMN.Game[]{PKMN.Game.RED}
                                           );

        PokemonTestCommon.TestCommon(pokemon, testParams);

        // Test attributes.

        Assert.AreEqual(pokemon.NumericAttributes["Catch rate"], 45);
    }
}

}

File: testing/unit-tests/pkmntest/lua/gen1_pokemon_tests.lua
--
-- Copyright (c) 2017-2018 Nicholas Corgan (n.corgan@gmail.com)
--
-- Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
-- or copy at http://opensource.org/licenses/MIT)
--

local pkmn = require("pkmn")
local luaunit = require("luaunit")
local pokemon_tests = require("pokemon_tests")

local gen1_pokemon_tests = {}

function gen1_pokemon_tests.common(game, species)
    local pokemon = pkmn.pokemon(species, game, "", 30)
    local test_params =
    {
        valid_ball = "Great Ball",
        invalid_balls = {"Great Ball"},

        valid_item = pkmn.item.POTION,
        invalid_items = {pkmn.item.POTION},

        expected_original_location = pkmn.stat.SPECIAL,
        valid_locations = {"Route 1"},
        invalid_locations = {"Route 1"},

        valid_moves = {"Slash", "Flamethrower", "Tail Whip", "Fire Blast"},
        invalid_moves = {"Return"},

        valid_original_games = {pkmn.game.RED},
        invalid_original_games = {pkmn.game.RED}
    }

    pokemon_tests.test_common(pokemon, test_params)

    luaunit.assertEquals(pokemon.numeric_attributes["Catch rate"], 45)
end

-- Red

function test_red_pokemon()
    gen1_pokemon_tests.common(pkmn.game.RED, pkmn.species.CHARMANDER)
end

function test_red_forms()
    pokemon_tests.forms_test(pkmn.game.RED)
end

-- Blue

function test_blue_pokemon()
    gen1_pokemon_tests.common(pkmn.game.BLUE, pkmn.species.SQUIRTLE)
end

function test_blue_forms()
    pokemon_tests.forms_test(pkmn.game.BLUE)
end

-- Yellow

function test_yellow_pokemon()
    gen1_pokemon_tests.common(pkmn.game.YELLOW, pkmn.species.BULBASAUR)
end

function test_yellow_forms()
    pokemon_tests.forms_test(pkmn.game.YELLOW)
end
